src/main/java/pixelperfect/Spaceship.java

```diff
@@ -15,6 +16,7 @@
 public class Spaceship {
 
 	private static final Spaceship ship = new Spaceship();
+	private RouteGenerator routeGenerator;
```

Comment:
The Route class will not become a data class: it will have features like:
- Getting a location from the route
- Calculating time that has past
- Doing things that are in the Route

Giving the RouteGenerator to Route seems a little bit odd. Giving a PizzaFactory to a Pizza means that actually you should delete the PizzaFactory and implement the factory in Pizza itself.